feat(component/tabs): implementation of the new design of the Tabs component

tiloyi opened the pull request:
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [x] Yes
- [ ] No

## Describe the changes

Implementation of the new design of the Tabs component.

GitHub issue number or Jira issue URL:
- Figma: https://www.figma.com/file/nx0yFysbE054fkLqLcnSL7/ADS---Specs-and-Docs?node-id=901%3A8110
- JIRA: https://jira.adeo.com/browse/MZC-267

caqueste reviewed (approved):
OK for me. Good job @tiloyi 
I deleted the design documentation which wasn't helpful and up-to-date.
However, I didn't find the new variation with the icon. It will be added in the next release 🙂

tiloyi commented:
> OK for me. Good job @tiloyi I deleted the design documentation which wasn't helpful and up-to-date. However, I didn't find the new variation with the icon. It will be added in the next release 🙂

Hi @caqueste , I just added the variation with icons. I let you check ;)

caqueste reviewed (approved):
:fire:
